journal/src/test/java/io/camunda/zeebe/journal/file/SegmentedJournalTest.java

```diff
@@ -652,6 +655,80 @@ void shouldNotPreallocateSegmentFiles(final @TempDir Path tmpDir) {
     PosixPathAssert.assertThat(firstSegment).hasRealSizeLessThan(segmentSize);
   }
 
+  @Test
+  void shouldValidateAsqnBeforeCreatingNewSegment() {
+    // given
+    // one entry fits but not two
+    final SegmentedJournal journal = openJournal(1.5f);
+    journal.append(1, data);
+
+    // when/then
+    Assertions.assertThatThrownBy(() -> journal.append(1, data)).isInstanceOf(InvalidASqn.class);
+    assertThat(journal.getFirstSegment()).isEqualTo(journal.getLastSegment());
+  }
+
+  @Test
+  void shouldValidateAsqnWhenWritingToNewSegment() {
+    // given
+    // one entry fits but not two
+    final SegmentedJournal journal = openJournal(1.5f);
+    journal.append(1, data);
+
+    // when
+    // force creation of new segment with an asqn ignore entry
+    journal.append(ASQN_IGNORE, data);
+
+    // then
+    // validation of the asqn should fail on the new segment as well
+    Assertions.assertThatThrownBy(() -> journal.append(1, data)).isInstanceOf(InvalidASqn.class);
+
+    assertThat(journal.getFirstSegment()).isNotEqualTo(journal.getLastSegment());
+  }
+
+  @Test
+  void shouldValidateAsqnWhenWritingAfterRestartOnSameSegment() {
+    // given
+    // one entry fits but not two
+    final float entriesPerSegment = 5;
+    SegmentedJournal journal = openJournal(entriesPerSegment);
+
+    journal.append(1, data);
+
+    // when
+    journal.close();
+    journal = openJournal(entriesPerSegment);
+
+    // then
+    final SegmentedJournal finalJournal = journal;
+    Assertions.assertThatThrownBy(() -> finalJournal.append(1, data))
+        .isInstanceOf(InvalidASqn.class);
+
+    assertThat(journal.getFirstSegment()).isEqualTo(journal.getLastSegment());
+  }
+
+  @Test
+  void shouldValidateAsqnWhenWritingAfterRestartOnNewSegmentWithOnlyAsqnIgnoreRecord() {
+    // given
+    // one entry fits but not two
+    final float entriesPerSegment = 1.5f;
+    SegmentedJournal journal = openJournal(entriesPerSegment);
+
+    journal.append(1, data);
+
+    // when
+    // force creation of new segment with an asqn ignore entry
+    journal.append(ASQN_IGNORE, data);
+    journal.close();
+    journal = openJournal(entriesPerSegment);
+
+    // then
+    final SegmentedJournal finalJournal = journal;
```

Comment:
:wrench: This may be unnecessary.

Reply:
it's here because of the following lambda in `assertThatThrownBy` that requires a final reference. However I went for just introducing:
```
    final SegmentedJournal reopenedJournal = openJournal(entriesPerSegment);
```

in the when blog when restarting.